python/doc/examples/getting_started/plot_kriging_vs_gpr.py

```diff
@@ -0,0 +1,507 @@
+# For that purpose, we create a Gaussian process Regression for a function which has scalar real inputs and outputs.
+# We select a very simple example .
+
+# %%
+# Introduction
+# ------------
+#
+# We consider the sine function:
+#
+# .. math::
+#    y = x \sin(x)
+#
+#
+# for any :math:`x\in[0,12]`.
+#
+# We want to create a surrogate of this function. This is why we create a sample of :math:`n` observations of the function:
+#
+# .. math::
+#    y_i=x_i \sin(x_i)
+#
+# We are going to consider a Gaussian Process Regression with:
+#
+# * a constant trend,
+# * a Matern covariance model.
+
+# %%
+import openturns as ot
+from openturns import viewer
+from matplotlib import pylab as plt
+import openturns.experimental as otexp
+
+ot.Log.Show(ot.Log.NONE)
+
+# %%
+# First let us introduce some useful function.
+# In order to observe the function and the location of the points in the input design of experiments, we define `plot_1d_data`.
+
+
+# %%
+def plot_1d_data(x_data, y_data, type="Curve", legend=None, color=None, linestyle=None):
+    """Plot the data (x_data,y_data) as a Cloud/Curve"""
+    if type == "Curve":
+        graphF = ot.Curve(x_data, y_data)
+    else:
+        graphF = ot.Cloud(x_data, y_data)
+    if legend is not None:
+        graphF.setLegend(legend)
+    if color is not None:
+        graphF.setColor(color)
+    if linestyle is not None:
+        graphF.setLineStyle(linestyle)
+    return graphF
+
+
+def computeQuantileAlpha(alpha):
+    """
+    Compute bilateral confidence interval of level 1-alpha of a gaussian distribution.
+    """
+    bilateralCI = ot.Normal().computeBilateralConfidenceInterval(1 - alpha)
+    return bilateralCI.getUpperBound()[0]
+
+
+def computeBoundsConfidenceInterval(y_test_hat, quantileAlpha, conditionalSigma):
+    """
+    Compute the 1-alpha confidence interval bounds.
+    """
+    dataLower = [
+        [y_test_hat[i, 0] - quantileAlpha * conditionalSigma[i, 0]]
+        for i in range(n_test)
+    ]
+    dataUpper = [
+        [y_test_hat[i, 0] + quantileAlpha * conditionalSigma[i, 0]]
+        for i in range(n_test)
+    ]
+    dataLower = ot.Sample(dataLower)
+    dataUpper = ot.Sample(dataUpper)
+    return dataLower, dataUpper
+
+
+# %%
+g = ot.SymbolicFunction(["x"], ["x * sin(x)"])
+
+# %%
+xmin = 0.0
+xmax = 12.0
+n_train = 20
+step = (xmax - 1 - xmin) / (n_train - 1.0)
+x_train = ot.RegularGrid(xmin + 0.2, step, n_train).getVertices()
+y_train = g(x_train)
+n_train = x_train.getSize()
+
+# %%
+# In order to compare the function and its metamodel, we use a test (i.e. validation) design of experiments made of a regular grid of 100 points from 0 to 12.
+# Then we convert this grid into a `Sample` and we compute the outputs of the function on this sample.
+
+# %%
+n_test = 100
+step = (xmax - xmin) / (n_test - 1)
+myRegularGrid = ot.RegularGrid(xmin, step, n_test)
+x_test = myRegularGrid.getVertices()
+y_test = g(x_test)
+
+# %%
+# We plot the test and train data
```

Comment:
In this plot, it lookslike the plot of the function (continuous curve) and not the test data (discrete points), so it is not really the plot of the design of experiments that is used for testing.